MAVProxy/modules/mavproxy_kmlread.py

```diff
@@ -68,19 +188,49 @@ def loadkml(self, filename):
             if self.mpstate.map is not None and point[0] == 'Polygon':
                 #print "Adding " + point[1]
                 newcolour = (random.randint(0, 255), 0, random.randint(0, 255))
-                self.mpstate.map.add_object(mp_slipmap.SlipPolygon(point[1], point[2],
-                                                             layer=2, linewidth=2, colour=newcolour))
+                curpoly = mp_slipmap.SlipPolygon(point[1], point[2],
+                                                             layer=2, linewidth=2, colour=newcolour)
+                self.mpstate.map.add_object(curpoly)
+                self.allayers.append(curpoly)
                 self.curlayers.append(point[1])
                 
                 
-            #and points - 20m circle
+            #and points - barrell image and text
             if self.mpstate.map is not None and point[0] == 'Point':
                 #print "Adding " + point[1]
                 icon = self.mpstate.map.icon('barrell.png')
-                self.mpstate.map.add_object(mp_slipmap.SlipIcon(point[1], latlon = (point[2][0][0], point[2][0][1]), layer=3, img=icon, rotation=0, follow=False))
+                curpoint = mp_slipmap.SlipIcon(point[1], latlon = (point[2][0][0], point[2][0][1]), layer=3, img=icon, rotation=0, follow=False)
+                curtext = mp_slipmap.SlipLabel(point[1], point = (point[2][0][0], point[2][0][1]), layer=4, label=point[1], colour=(0,255,255))
+                self.mpstate.map.add_object(curpoint)
+                self.mpstate.map.add_object(curtext)
+                self.allayers.append(curpoint)
+                self.alltextlayers.append(curtext)
                 self.curlayers.append(point[1])
-                
-        
+                self.curtextlayers.append(point[1])                
+
+    def idle_task(self):
+        '''handle GUI elements'''
+        if self.module('map') is not None and not self.menu_added_map:
+            self.menu_added_map = True
+            self.module('map').add_menu(self.menu)
+        #(re)create the menu
+        if mp_util.has_wxpython and self.menu_added_map:
+            # we don't dynamically update these yet due to a wx bug
```

Comment:
I don't think all this needs to be done on every call of idle_task(). That seems to just waste cpu